drivers/can/socket_can_generic.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+/* Generic CANBUS related functions that are generic in all the drivers. */
+
+#ifndef ZEPHYR_DRIVERS_CAN_SOCKET_CAN_GENERIC_H_
+#define ZEPHYR_DRIVERS_CAN_SOCKET_CAN_GENERIC_H_
+
+#define SOCKET_CAN_NAME_1 "SOCKET_CAN_1"
+#define SEND_TIMEOUT K_MSEC(100)
+#define RX_THREAD_STACK_SIZE 512
+#define RX_THREAD_PRIORITY 2
+#define BUF_ALLOC_TIMEOUT K_MSEC(50)
+
+CAN_DEFINE_MSGQ(socket_can_msgq, 5);
+K_THREAD_STACK_DEFINE(rx_thread_stack, RX_THREAD_STACK_SIZE);
+
+struct socket_can_context {
+	struct device *can_dev;
+	struct net_if *iface;
+	struct k_msgq *msgq;
+	k_tid_t rx_tid;
+	struct k_thread rx_thread_data;
+	struct can_filter filter;
```

Comment:
Why do you need a filter in socket_can_context?

Reply:
I was not sure what the CAN driver will do with the filter. If the driver just copies the filter or does not refer to the original buffer, then sure the extra variable is not needed. I will remove it.

Reply:
The filter is just a generic struct that represents filters. The driver takes this information and sets the registers accordingly. Its save to discard after the call.